tensorflow/compiler/tf2tensorrt/common/utils.h

```diff
@@ -54,6 +56,64 @@ void MaybeInitializeTrtPlugins(nvinfer1::ILogger* trt_logger);
 }  // namespace tensorrt
 }  // namespace tensorflow
 
+namespace nvinfer1 {
+
+// Alias a useful type computation for nvinfer1::Dims types. Types like
+// nvinfer1::Dims2 inherit from nvinfer1::Dims.
```

Comment:
I think we should remove the support for the legacy DimsN for two reasons: We don't actually use it in the TF-TRT converter. I see a few use in the test, but that can be fixed; DimsN is a convenient way of defining Dims (fix the nbDims to N) and will be deprecated in TensorRT. If we only support Dims NOT support DimsN, will this simplify the code here?

Reply:
In a later PR I can remove deprecated type usages from the tests, and then remove this SFINAE template usage. 

If I understand what you're saying, we should have utilities called DimsN, which would just be functions which create an `nvinfer1::Dims`?

In follow on [PR I moved utility functions](https://github.com/tensorflow/tensorflow/pull/51435/files#diff-bd883d225ea57894fd8bbe9160ac8cc3233c1947bb532fb544c309d7dcaecb03R138-R154) under `common/utils.h` for nvinfer1::Dims creation; I can put the DimsN creation utilities there later when I remove the deprecated types.

Reply:
Can you explain why you need to use SFINAE here?
What would be the problem if you just define std::ostream& operator<<(std::ostream& os, const nvinfer1::Dims& v) in this PR instead of introducing SFINAE and then remove it later? 
We don't actually use DimsN in the converter. But even if we use DimsN, std::ostream& operator<<(std::ostream& os, const nvinfer1::Dims& v) should also works for DimsN, am I right here?

Reply:
Yes it works that way as well. I just like this template technique because you can apply it to to things like non-ref values and the value type of a vector, etc. I feel like it tends to come up a lot in header files like this for stream operators and looks cleaner when they all have template definition. Up to you, I'm fine either way

Reply:
We only use SFINAE when it is necessary. Please simplify this code and remove SFINAE then.